code/modules/wiremod/integrated_circuit.dm

```diff
@@ -48,12 +48,24 @@ GLOBAL_LIST_EMPTY_TYPED(integrated_circuits, /obj/item/integrated_circuit)
 	/// Set by the shell. Holds the reference to the owner who inserted the component into the shell.
 	var/datum/weakref/inserter_mind
 
+	/// Variables stored on this integrated circuit. with a `variable_name = value` structure
+	var/list/datum/circuit_variable/circuit_variables = list()
+
+	var/max_setters_and_getters = 30
+	var/setter_and_getter_count = 0
+
 	/// X position of the examined_component
 	var/examined_rel_x = 0
 
 	/// Y position of the examined component
 	var/examined_rel_y = 0
 
+	/// The X position of the screen. Used for adding components
+	var/screen_x = 0
```

Comment:
Why is stuff like this sent to server now? I'm worried

Reply:
explained on discord:
positioning of new components depend on the last screen_x and screen_y sent to the server. Additionally, when the UI is re-opened, the screen will also be re-positioned here.